src/vm/instruction.ts

```diff
@@ -27,40 +27,130 @@ import {
 // └─────┴─────┴───┴───┴───┴───┴───┴───┴───┴───┴────┴────┴────┴────┴────┴────┘
 // Source: https://github.com/lambdaclass/cairo-vm_in_go/blob/main/pkg/vm/instruction.go
 
-// Dst & Op0 register flags
-export type RegisterFlag = 'ap' | 'fp';
+/** Flags to compute `dst` and `op0` from a register */
+export enum Register {
+  /**
+   * - dst = [ap + dstOffset]
+   * - op0 = [ap + op0Offset]
+   */
+  Ap,
+  /**
+   * - dst = [fp + dstOffset]
+   * - op0 = [fp + op0Offset]
+   */
+  Fp,
+}
 
-export type Op1Source = 'op0' | 'pc' | 'fp' | 'ap';
+/**
+ * Source from which op1 is computed
+ *
+ * op1 = [op1Source + op1Offset]
+ */
+export enum Op1Source {
```

Comment:
There is an issue regarding this #40 so maybe ok to keep this as is right not

Reply:
Yes, I planned on tackling this issue in a next PR, while refactoring the `computeOperands()` and related methods (#54 )